src/entries/popup/pages/home/TokenDetails/TokenContextMenu.tsx

```diff
@@ -130,19 +135,11 @@ export function TokenContextMenu({ children, token }: TokenContextMenuProps) {
     toggleHideAsset(address, computeUniqueIdForHiddenAsset(token));
     if (pinned) togglePinAsset(address, token.uniqueId);
     setSelectedToken();
-    triggerToast({
-      title: i18n.t('token_details.toast.hide_token', {
-        name: token.symbol,
-      }),
-    });
-    const isHidden =
-      useHiddenAssetStore.getState().hidden[address]?.[
-        computeUniqueIdForHiddenAsset(token)
-      ];
-    const hiddenCount = Object.values(
-      useHiddenAssetStore.getState().hidden[address] || {},
-    ).filter((isHidden) => isHidden).length;
-    analytics.track(isHidden ? event.tokenHidden : event.tokenUnhidden, {
+    const newIsHidden = !hidden;
+    const hiddenCount =
+      Object.values(hiddenStore[address] || {}).filter((isHidden) => isHidden)
+        .length + (newIsHidden ? 1 : -1);
+    analytics.track(newIsHidden ? event.tokenHidden : event.tokenUnhidden, {
```

Comment:
Would inline the newIsHidden here; also generally would call the toast before the analytics in the unlikely case that the analytics call is blocking